rust-sdk/utils/src/math/tick.rs

```diff
@@ -0,0 +1,349 @@
+pub fn get_next_initializable_tick_index(
+    tick_index: TickIndex,
+    tick_spacing: TickSpacing,
+) -> TickIndex {
+    let tick_spacing_i32 = tick_spacing as i32;
+    get_prev_initializable_tick_index(tick_index, tick_spacing) + tick_spacing_i32
+}
+
+/// Check if a tick is in-bounds.
+///
+/// # Parameters
+/// - `tick_index` - A TickIndex integer representing the tick integer
+///
+/// # Returns
+/// - A boolean value indicating if the tick is in-bounds
+pub fn is_tick_index_in_bounds(tick_index: TickIndex) -> bool {
+    tick_index <= MAX_TICK_INDEX && tick_index >= MIN_TICK_INDEX
+}
+
+/// Check if a tick is initializable.
+/// A tick is initializable if it is divisible by the tick spacing.
+///
+/// # Parameters
+/// - `tick_index` - A TickIndex integer representing the tick integer
+/// - `tick_spacing` - A TickIndex integer representing the tick spacing
+///
+/// # Returns
+/// - A boolean value indicating if the tick is initializable
+pub fn is_tick_initializable(tick_index: TickIndex, tick_spacing: TickSpacing) -> bool {
+    let tick_spacing_i32 = tick_spacing as i32;
+    tick_index % tick_spacing_i32 == 0
+}
+
+/// Get the tick index for the inverse of the price that this tick represents.
+/// Eg: Consider tick i where Pb/Pa = 1.0001 ^ i
+/// inverse of this, i.e. Pa/Pb = 1 / (1.0001 ^ i) = 1.0001^-i
+///
+/// # Parameters
+/// - `tick_index` - A TickIndex integer representing the tick integer
+///
+/// # Returns
+/// - A TickIndex integer representing the tick index for the inverse of the price
+pub fn invert_tick_index(tick_index: TickIndex) -> TickIndex {
+    -tick_index
+}
+
+/// Get the minimum and maximum tick index that can be initialized.
+///
+/// # Parameters
+/// - `tick_spacing` - A TickIndex integer representing the tick spacing
+///
+/// # Returns
+/// - A TickRange struct containing the lower and upper tick index
+pub fn get_full_range_tick_indexes(tick_spacing: TickSpacing) -> TickRange {
+    let tick_spacing_i32 = tick_spacing as i32;
+    let min_tick_index = (MIN_TICK_INDEX / tick_spacing_i32) * tick_spacing_i32;
+    let max_tick_index = (MAX_TICK_INDEX / tick_spacing_i32) * tick_spacing_i32;
+    TickRange {
+        lower: min_tick_index,
+        upper: max_tick_index,
+    }
+}
+
+pub fn is_full_range_only(tick_spacing: TickSpacing) -> bool {
+    tick_spacing >= FULL_RANGE_ONLY_TICK_SPACING_THRESHOLD
+}
+
+//   /**
+//    * Check if a whirlpool is a full-range only pool.
+//    * @param tickSpacing The tickSpacing for the Whirlpool
+//    * @returns true if the whirlpool is a full-range only pool, false otherwise.
+//    */
+//   public static isFullRangeOnly(tickSpacing: number): boolean {
+//     return tickSpacing >= FULL_RANGE_ONLY_TICK_SPACING_THRESHOLD;
+//   }
+
+// Private functions
+
+fn mul_shift_96(n0: u128, n1: u128) -> u128 {
+    let mul = <U256>::from(n0) * <U256>::from(n1);
+    mul.wrapping_shl(96).try_into().unwrap()
+}
+
+fn get_sqrt_price_positive_tick(tick: TickIndex) -> u128 {
```

Comment:
might be good to link to where these constants come from and/or describe the overall algorithm in a comment